lib/core/src/Cardano/Wallet.hs

```diff
@@ -720,6 +738,21 @@ newWalletLayer tracer bp db nw tl = do
             (w, _) <- _readWallet wid
             DB.putCheckpoint db (PrimaryKey wid) (newPending (tx, meta) w)
 
+    _slotIdTime
+        :: SlotId
+        -> UTCTime
+    _slotIdTime slN =
+        addUTCTime (convert $ slotLength * slots) block0Date
+      where
+        slots = fromIntegral . fromEnum $ (flatSlot' slN) - (flatSlot' sl0)
+        sl0 = block0 ^. #header ^. #slotId
```

Comment:
I don't think this should live in the `WalletLayer`. This is a pure function that could live in the `Primitive.Types` module.

Reply:
Tbh, I'm thinking it might be best to go for the full solution (with dynamic slotting) straight away.

Reply:
Seems reasonable. You don't need to make many assumption about how the slot times are determined in order to implement this function. You can expect an index mapping `SlotNo` (or `(Epoch, SlotId)` to a given `(SlotLength, EpochLength)` and make sure to correctly compute the corresponding time. 

I'd even expect this to be so isolated that it could live in its own PR with dedicated testing. No need to do any modification to the wallet layer to implement this.

Reply:
Or no, I changed my mind 😅 .

Storing a list, `[SlottingConfiguration]`,
```haskell
data SlottingConfiguration 
   { validFrom :: SlotId
    , slotLength :: SlotLength
    , epochLength :: EpochLength
    }
```
would be kind of plausible assuming the slotting configuration doesn't change regularly, but for _me_, the same fears as here https://github.com/input-output-hk/cardano-wallet/pull/563#discussion_r304824349 would apply.